yt_dlp/extractor/twitter.py

```diff
@@ -503,90 +607,114 @@ def add_thumbnail(name, size):
+            # video_direct_message, poll2choice_video, poll3choice_video,
+            # poll4choice_video, ...
+            else:
+                is_amplify = card_name == 'amplify'
+                vmap_url = get_binding_value('amplify_url_vmap') if is_amplify else get_binding_value('player_stream_url')
+                content_id = get_binding_value('%s_content_id' % (card_name if is_amplify else 'player'))
+                formats, subtitles = self._extract_formats_from_vmap_url(vmap_url, content_id or twid)
+                self._sort_formats(formats)
+
+                thumbnails = []
+                for suffix in ('_small', '', '_large', '_x_large', '_original'):
+                    image = get_binding_value('player_image' + suffix) or {}
+                    image_url = image.get('url')
+                    if not image_url or '/player-placeholder' in image_url:
+                        continue
+                    thumbnails.append({
+                        'id': suffix[1:] if suffix else 'medium',
+                        'url': image_url,
+                        'width': int_or_none(image.get('width')),
+                        'height': int_or_none(image.get('height')),
+                    })
+
+                return {
+                    'formats': formats,
+                    'subtitles': subtitles,
+                    'thumbnails': thumbnails,
+                    'duration': int_or_none(get_binding_value(
+                        'content_duration_seconds')),
+                }
+
+        entries = []
+        media_path = ((None, 'quoted_status'), 'extended_entities', 'media', ...)
+        medias = traverse_obj(status, media_path, expected_type=dict, default=[])
+        for media in medias:
+            if not media or media.get('type') == 'photo':
+                continue
+
+            data = extract_from_video_info(media)
+            if data:
+                entries.append({**info, **data, 'display_id': twid})
+            continue
+
+        card = status.get('card')
+        if card:
+            data = extract_from_card_info(card)
+            if data:
+                entries.append({**info, **data, 'display_id': twid})
+
+        if not entries:
+            expanded_url = traverse_obj(status, ('entities', 'urls', 0, 'expanded_url'))
+            if not expanded_url or expanded_url == url:
+                raise ExtractorError('No video could be found in this tweet')
+
+            transparent_result = self.url_result(expanded_url, transparent=True, display_id=twid, **info)
+            if not url_or_none(transparent_result.get('url')) and not transparent_result.get('formats'):
+                raise ExtractorError('No video could be found in this tweet')
+
+            return transparent_result
+
+        if len(entries) == 1:
+            # compat: tweet id was previously used instead of media id
+            return {**entries[0], 'id': twid}
```

Comment:
Another way to handle compat is to set `_old_archive_ids`

Reply:
Though this would still break compat since the id field changed. This way you also still keep reference to the main tweet id in the result directly. Whichever you prefer.

Reply:
Then I think using the new `id` is more accurate.

Reply:
In case it wasn't obvious, I set the `id` for each of multiple videos from the `id_str` in the API response (and the `title` had `#{n}` appended) to make them distinct, but the original `title` and `id` were restored if only one video was found, giving the original result. In the multiple case, the playlist retained the original `id` and would be the `playlist_id` in the output JSON.

Reply:
What is your reasoning behind restoring the old `id`? I find it beneficial to provide the actual `media_id` instead, since thats consistent with the behavior of the playlist variation.

Changing the title to reflect media index makes a lot of sense. I will implement that as well.

Reply:
> What is your reasoning behind restoring the old id?

Likely because youtube-dl does not have `_old_archive_ids`

Reply:
Just that I wanted to return the same `id` as would have been returned by the single video extractor when the API response had a video in `status['extended_entities']['media'][0]`.

I also trimmed the title to 120 characters, but that's another issue.

Reply:
Since we have `_old_archive_ids` and `id` matching does not seem to be a concern I would rather return the medias id instead.